ebay-radio: TypeScript compilation error with onChange

Similar to issue #75, I got TypeScript compilation error when using EbayCheckbox with onChange:

```
error TS2322: Type '(e: ChangeEvent<HTMLInputElement>, value: string | number, checked: boolean) => void' is not assignable to type 'ChangeEventHandler<HTMLInputElement> & ((e: ChangeEvent<HTMLInputElement>, value: string | number, checked: boolean) => void)'.
  Type '(e: ChangeEvent<HTMLInputElement>, value: string | number, checked: boolean) => void' is not assignable to type 'ChangeEventHandler<HTMLInputElement>'.

68       <EbayCheckbox checked value="123" id="checkbox-21" onChange={handleCheckboxChange}>
                                                            ~~~~~~~~

  node_modules/@types/react/index.d.ts:2221:9
    2221         onChange?: ChangeEventHandler<T> | undefined;
                 ~~~~~~~~
    The expected type comes from property 'onChange' which is declared here on type 'IntrinsicAttributes & InputProps & EbayCheckboxProps & { children?: ReactNode; }'
```

## How to reproduce

- Use any TypeScript (tested with 3.9 and 4.6) with the latest type definitions for react v16.14.30.
- Add EbayCheckbox with onChange

```tsx
const handleCheckboxChange: (
  e: React.ChangeEvent<HTMLInputElement>, value: string | number, checked: boolean
) => void = (event, selectedIndex, newValue) => {
   // ...
}

// ...
<EbayCheckbox checked value="123" id="checkbox-21" onChange={handleCheckboxChange}>
  <EbayLabel>Default</EbayLabel>
</EbayCheckbox>
```

## Cause

This is due to conflicting types between:

- HTML input onChange with type `ChangeEventHandler<HTMLInputElement>` (1 parameter)
- EbaySelect onChange with type `(e: ChangeEvent<HTMLInputElement>, value: string | number, checked: boolean) => void` (3 parameters)

EbayCheckbox used all `<input>` props and try to overwrite its onChange: https://github.com/eBay/ebayui-core-react/blob/e13c4e6f9beba01ce19ba8c2c94d77c3dec57357/src/ebay-checkbox/checkbox.tsx#L8-L12

We should omit onchange from EbayCheckbox before overwriting it. See https://react-typescript-cheatsheet.netlify.app/docs/advanced/patterns_by_usecase/#props-omit-prop-from-a-type

## Workaround

Cast the handler with ChangeEventHandler<HTMLSelectElement>:

```tsx
<EbayCheckbox
  onChange={handleCheckboxChange as React.ChangeEventHandler<HTMLInputElement>}
>
```

--- a/src/ebay-checkbox/__tests__/index.stories.tsx
+++ b/src/ebay-checkbox/__tests__/index.stories.tsx
@@ -1,4 +1,4 @@
-import React, { useState, useRef, ReactElement } from 'react'
+import React, { useState, useRef, ReactElement, ChangeEvent } from 'react';
 import { storiesOf } from '@storybook/react'
 import { action } from '../../../.storybook/action'
 import { EbayCheckbox, EbayLabel } from '../../index'
@@ -97,9 +97,9 @@ storiesOf(`ebay-checkbox`, module)
             const [isChecked, setChecked] = useState(true)
             const [isDisabled, setDisabled] = useState(false)
             const counter = useRef(0)
-            const handleOnChange = (e) => {
+            const handleOnChange = (e: ChangeEvent<HTMLInputElement>, value: string | number, checked: boolean) => {
                 if (counter.current < 4) {
-                    setChecked(e.target.checked)
+                    setChecked(checked)
                 } else {
                     setDisabled(true)
                 }

--- a/src/ebay-checkbox/checkbox.tsx
+++ b/src/ebay-checkbox/checkbox.tsx
@@ -5,7 +5,7 @@ import { EbayLabel, EbayLabelProps } from '../ebay-field'
 import { findComponent } from '../common/component-utils'
 
 type Size = 'default' | 'large'
-type InputProps = Omit<ComponentProps<'input'>, 'size'>
+type InputProps = Omit<ComponentProps<'input'>, 'size' | 'onChange'>
 type EbayCheckboxProps = {
     size?: Size;
     onChange?: (e: ChangeEvent<HTMLInputElement>, value: string | number, checked: boolean) => void;

